R/sampleSummaryStats.R

```diff
@@ -1,134 +1,140 @@
-#' @export
-setGeneric("sampleSummaryStatsTable", function(inSCE, ...) standardGeneric("sampleSummaryStatsTable"))
-
-#' @export
-setGeneric("sampleSummaryStatsTable<-", function(inSCE, ..., value) standardGeneric("sampleSummaryStatsTable<-"))
-
-#' @export
+#' @rdname sampleSummaryStatsTable
 setMethod("sampleSummaryStatsTable", "SingleCellExperiment", function(inSCE, statsName, ...){
-    return(inSCE@metadata$sctk$sampleSummary[[statsName]])
+    allStatsNames <- listSampleSummaryStatsTables(inSCE)
+    if(length(allStatsNames) == 0){
+        stop(paste("No tables are stored within the SingleCellExperiment object."),
+             "Please execute the sampleSummaryStats function first.")
+    }
+    if(!statsName %in% allStatsNames){
+        stop(paste(statsName,
+                   "is not a table within the SingleCellExperiment object.",
+                   "The following are the names of the tables stored:",
+                   allStatsNames))
+    }else{
+        return(inSCE@metadata$sctk$sampleSummary[[statsName]])
+    }
 })
 
-#' @rdname sampleSummaryStatsTable
-#' @title Stores and returns table of SCTK QC outputs to metadata.
-#' @description  Stores and returns table of QC metrics generated from
-#'  QC algorithms within the metadata slot of the SingleCellExperiment object.
-#' @param inSCE Input \linkS4class{SingleCellExperiment} object with saved
-#' \link{assay} data and/or \link{colData} data. Required.
-#' @param statsName A \code{character} value indicating the slot
-#' that stores the stats table within the metadata of the
-#' SingleCellExperiment object. Required.
-#' @return A matrix/array object. Contains a summary table for QC statistics
-#' generated from SingleCellTK.
-#' @examples
-#' data(scExample, package = "singleCellTK")
-#' sce <- subsetSCECols(sce, colData = "type != 'EmptyDroplet'")
-#' sce <- sampleSummaryStats(sce, simple = TRUE)
-#' sampleSummaryStatsTable(sce, statsName = "sctk_qc")
-#' @export
+#' @rdname sampleSummaryStatsTable<-
 setReplaceMethod("sampleSummaryStatsTable", c("SingleCellExperiment", "ANY"), function(inSCE, statsName, ..., value) {
     inSCE@metadata$sctk$sampleSummary[[statsName]] <- value
     return(inSCE)
 })
 
-.sampleSummaryStats <- function(inSCE, colName = "Total",
+#' @rdname listSampleSummaryStatsTables
+setMethod("listSampleSummaryStatsTables", "SingleCellExperiment", function(inSCE, statsName, ...){
+    if(is.null(inSCE@metadata$sctk$sampleSummary)){
+        return(paste("No tables found in the SingleCellExperiment object.",
+                     "Please execute the sampleSummaryStats function first."))
+    }else{
+        allStatsNames <- names(inSCE@metadata$sctk$sampleSummary)
+        if(length(allStatsNames) == 0){
```

Comment:
if this hasn't been set before then I would expect allStatsNames to be NULL, not a list of length zero. Maybe a better check would be is.null(allStatsName) || length(allStatsNames == 0). A double "||" will check for NULL first and then only do the second check if the first one passes.